independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/AnalyticsService.java

```diff
@@ -0,0 +1,260 @@
+                .pair(PROP_VERSION, getProperty("java.version", "N/A"))
+                .build()
+                .mapPair(PROP_GRAALVM)
+                .pair(PROP_VENDOR, ofNullable(buildInfo.get(GRAALVM_VERSION_DISTRIBUTION)).orElse("N/A"))
+                .pair(PROP_VERSION, ofNullable(buildInfo.get(GRAALVM_VERSION_VERSION)).orElse("N/A"))
+                .pair(PROP_JAVA_VERSION, ofNullable(buildInfo.get(GRAALVM_VERSION_JAVA)).orElse("N/A"))
+                .build()
+                .mapPair(PROP_BUILD)
+                .pair(PROP_MAVEN_VERSION, ofNullable(buildInfo.get(MAVEN_VERSION)).orElse("N/A"))
+                .pair(PROP_GRADLE_VERSION, ofNullable(buildInfo.get(GRADLE_VERSION)).orElse("N/A"))
+                .build()
+                .mapPair(PROP_QUARKUS)
+                .pair(PROP_VERSION, getQuarkusVersion(applicationModel))
+                .build()
+                .pair(PROP_IP, VALUE_NULL_IP)
+                .mapPair(PROP_LOCATION)
+                .pair(PROP_COUNTRY, Locale.getDefault().getCountry())
+                .build()
+                .mapPair(PROP_OS)
+                .pair(PROP_NAME, getProperty("os.name", "N/A"))
+                .pair(PROP_VERSION, getProperty("os.version", "N/A"))
+                .pair(PROP_OS_ARCH, getProperty("os.arch", "N/A"))
+                .build()
+                .mapPair(PROP_CI)
+                .pair(PROP_CI_NAME, getBuildSystemName())
+                .build()
+                .mapPair(PROP_KUBERNETES)
+                .pair(PROP_DETECTED, isKubernetesDetected())
+                .build()
+                .pair(PROP_TIMEZONE, ZoneId.systemDefault().getDisplayName(TextStyle.NARROW, Locale.ENGLISH))
+                .build();
+    }
+
+    private String isKubernetesDetected() {
+        return Boolean.toString(allEnvSet(
+                "KUBERNETES_SERVICE_HOST",
+                "KUBERNETES_SERVICE_PORT"));
+    }
+
+    private String getBuildSystemName() {
+        String travis = getenv("TRAVIS");
+        String user = getenv("USER");
+        if ("true".equals(travis) && "travis".equals(user)) {
+            return "travis";
+        }
+
+        if (allEnvSet("JENKINS_URL", "JENKINS_HOME", "WORKSPACE")) {
+            return "jenkins";
+        }
+
+        if (allEnvSet("GITHUB_WORKFLOW", "GITHUB_WORKSPACE", "GITHUB_RUN_ID")) {
+            return "github-actions";
+        }
+
+        // https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml
+        if (allEnvSet("BUILD_REASON", "AGENT_JOBSTATUS")) {
+            return "azure-pipelines";
+        }
+
+        return "unknown";
+    }
+
+    private boolean allEnvSet(String... names) {
+        for (String name : names) {
+            if (getenv(name) == null) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private String getQuarkusVersion(ApplicationModel applicationModel) {
```

Comment:
Is it worth to check whether the "plain" `quarkus-bom` or the platform `quarkus-bom` is used?
IIUC it's recommended to use the platform BOM - so that information feels useful.

Reply:
yes, I believe we should distinguish between them. Probably we should do it in a follow up PR.